certify-core/src/main/java/io/mosip/certify/core/util/AuditHelper.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package io.mosip.certify.core.util;
+
+import io.mosip.certify.api.dto.AuditDTO;
+import io.mosip.certify.core.dto.OIDCTransaction;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+public class AuditHelper {
+
+    public static AuditDTO buildAuditDto(String clientId) {
+        AuditDTO auditDTO = new AuditDTO();
+        auditDTO.setClientId(clientId);
+        auditDTO.setTransactionId(clientId);
+        auditDTO.setIdType("ClientId");
+        return auditDTO;
+    }
+
+    public static AuditDTO buildAuditDto(String transactionId, OIDCTransaction transaction) {
+        return buildAuditDto(transactionId, "transaction", transaction);
+    }
+
+    public static AuditDTO buildAuditDto(String transactionId, String idType, OIDCTransaction transaction) {
+        AuditDTO auditDTO = new AuditDTO();
+        auditDTO.setTransactionId(transactionId);
+        auditDTO.setIdType(idType);
+        if(transaction != null) {
+            auditDTO.setRelyingPartyId(transaction.getRelyingPartyId());
```

Comment:
Some of the attributes in the AuditDTO might not apply to the certify-service. Namely, relyingPartyId, claims related attributes, linkcode related attributes

Reply:
removed non applicable fields